java.source.base/src/org/netbeans/api/java/source/TreeUtilities.java

```diff
@@ -1883,6 +1883,51 @@ public boolean isEndOfCompoundVariableDeclaration(@NonNull Tree tree) {
         return false;
     }
  
+    /**
+     * Checks whether tree is part of compound variable declaration.
+     * {@link Tree}.
+     *
+     * @param tree the tree{@link Tree}
+     * @return true if tree is part of compound variable declaration else return
+     * false.
+     */
+    public boolean isPartOfCompoundVariableDeclaration(@NonNull Tree tree) {
+        TokenSequence<JavaTokenId> tokenSequence = tokensFor(tree);
+
+        if (tree.getKind() != Tree.Kind.VARIABLE) {
+            return false;
+        }
+
+        // If tree ends with comma then tree is part of compound variable declaration.
+        tokenSequence.moveEnd();
+        if (tokenSequence.movePrevious() && tokenSequence.token().id() == JavaTokenId.COMMA) {
```

Comment:
I didn't try - sorry. But will it work if there's a whitespace between identifier and the following comma: `int i = 10 , j = 1` ?

Reply:
This scenario is also working. I had tested .
 For subtree j=1, it  checks whether subtree (int i=10) ends with comma or not.

Reply:
:+1: